### PR TITLE
[ICD]Post the MsgSentEvent on retransmits also

### LINKED ISSUE #27864: [1.2] [ICD] MRP Retransmit of message should also trigger the active mode
### Reproduction steps

This only impacts an ICD server. 
A message sent event triggers or, possibly, extends the Active mode interval. But the MRP retransmit mechanism doesn't trigger that message sent event or impact the Active mode interval.

A retransmit should impact the ICD modes just as the initial message sent does.

### Bug prevalence

Always for and ICD

### GitHub hash of the SDK that was being used

ICD Manager not yet merged (will update once merge)

### Platform

core

### Platform Version(s)

_No response_

### Type

Core SDK Performance Improvement

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
This will kick the ICD into Active mode or extend its active mode duration for retransmitting messages also.
Fixes #27864 

Also, remove a noisy trace (particularly when idle/active interval are shorts)

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: src/messaging/ReliableMessageMgr.cpp
Comment (jmartinez-silabs) [reply src/messaging/ReliableMessageMgr.cpp:329]: Right, the initial message set the expected response(if needed) and the ICD will be waiting for the response or the timeout to unblock. 

I'll add this detail in comment format. Thanks!

**Review by lpbeliveau — APPROVED** [review]

**Pull request merged**
